tests/test_autonomy/test_configurations.py

```diff
@@ -139,3 +144,27 @@ def teardown(
 
         os.chdir(cls.cwd)
         shutil.rmtree(cls.t)
+
+
+@pytest.mark.parametrize(
+    "service_file",
+    Path(
+        ROOT_DIR,
+        "tests",
+        "data",
+        "dummy_service_config_files",
+    ).iterdir(),
+)
+def test_load_service(service_file: Path) -> None:
+    """Test loading and processing service component."""
+
+    with open_file(service_file, "r", encoding="utf-8") as fp:
+        data = yaml_load_all(fp)
+
+    data = cast(List[Dict], apply_env_variables(data, env_variables=os.environ))
```

Comment:
seeing this makes me wonder: how do we ensure we clean up environmental variables after a test?

Reply:
We don't need to, `apply_env_variables ` does not change anything in the `os.environ` neither we do for any kind of change